lib/modules/manager/npm/update/dependency/pnpm.ts

```diff
@@ -0,0 +1,104 @@
+import is from '@sindresorhus/is';
+import { stringify } from 'yaml';
+import { logger } from '../../../../../logger';
+import { parseSingleYamlDocument } from '../../../../../util/yaml';
+import type { UpdateDependencyConfig } from '../../../types';
+import { pnpmCatalogsSchema } from '../../extract/pnpm';
+import { getNewGitValue, getNewNpmAliasValue } from './common';
+
+export function updatePnpmCatalogDependency({
+  fileContent,
+  upgrade,
+}: UpdateDependencyConfig): string | null {
+  const { depType, managerData, depName } = upgrade;
+
+  const catalogName = managerData?.catalogName;
+
+  if (!is.string(catalogName)) {
+    logger.error(
+      'No catalogName was found; this is likely an extraction error.',
+    );
+    return null;
+  }
+
+  let { newValue } = upgrade;
+
+  newValue = getNewGitValue(upgrade) ?? newValue;
+  newValue = getNewNpmAliasValue(newValue, upgrade) ?? newValue;
+
+  logger.debug(
+    `npm.updatePnpmCatalogDependency(): ${depType}:${managerData?.catalogName}.${depName} = ${newValue}`,
+  );
+
+  let document;
+  let parsedContents;
+
+  try {
+    document = parseSingleYamlDocument(fileContent);
+    parsedContents = pnpmCatalogsSchema.parse(document.toJS());
```

Comment:
I opted to do the YAML replacements with a document representation, instead of string replacements, because it seemed simpler as a start. I particularly did not want to drill down into the catalogs, accounting for the different YAML Map representations, string quoting etc. 

Now that the tests are relatively set, I could change the implementation to do string replacement, if you think that is best.

Reply:
Does your solution keep whitespaces and comments?  

If yes, you can keep it like it is.